R/regression.R

```diff
@@ -1241,16 +1236,45 @@ fitModel <- function(.formula, .estimation.data, .weights, type, robust.se, subs
                                                family = poisson()),
                             "Quasi-Poisson" = svyglm(.formula, .design, subset = non.outlier.data_GQ9KqD7YOf,
                                                      family = quasipoisson()))
-            assign(".design", .design, envir=.GlobalEnv)
+            env <- environment(model[["formula"]])
+            assign(".design", .design, envir = env)
+            assign("svyglm", svyglm, envir = env)
+            on.exit({
+                remove(".design", envir = env)
+                remove("svyglm", envir = env)
+            })
             aic <- extractAIC(model)
-            remove(".design", envir=.GlobalEnv)
             model$df <- aic[1]
             model$aic <- aic[2]
         }
     }
     return(list(model = model, formula = .formula, design = .design, estimation.data = .estimation.data))
 }
 
+# survey (4.1.1) AIC extraction function removes intercept terms causing singularity in
+# intercept only models. Interim fix below which computes the AIC information properly
+#' @importFrom survey svyvar
```

Comment:
Add link to reference potentially referring to particular equations?

Reply:
Added some inline comments in ee2114e472d7596faf82392bad1d2867346701dd